app/helpers/form_helper.rb

```diff
@@ -354,7 +354,8 @@ def link_to_add_fields(name, f, association, partial = nil, options = {})
         { :f => builder }.merge(locals_option))
     end
     options[:class] = link_to_add_fields_classes(options)
-    link_to_function(name, "add_fields('#{options[:target]}', '#{association}', '#{escape_javascript(fields)}')".html_safe, options)
+    direction = ", '#{options[:direction] || 'append'}'"
+    link_to_function(name, "add_fields('#{options[:target]}', '#{association}', '#{escape_javascript(fields)}'#{direction})".html_safe, options)
```

Comment:
I had a hard time to figure out what the direction really does, it felt like you're joining it to the fields, given we are always including it, I think this is much more readable:
```suggestion
    link_to_function(name, "add_fields('#{options[:target]}', '#{association}', '#{escape_javascript(fields)}', #{options[:direction] || 'append'})".html_safe, options)
```

Reply:
ah right, totally makes sense, accepting